core/modules/replica/DatabaseMySQL.h

```diff
@@ -838,6 +568,84 @@ class Connection
+     *     } catch (ConnectTimeout const& ex) {
+     *         cerr << "you have exausted the maximum allowed number of retries "
+     *              << "within the specified (or implicitly assumed) "
+     *                 "timeout: " << ex.timeoutSec() << endl;
+     *     } catch (Error const& ex) {
+     *         cerr << "failed due to an unrecoverable error: " << ex.what() 
+     *     }
+     *   @code
+     *
+     *
+     * @param script        - user-provided function (the callable) to execute
+     * @param maxReconnects - (optional) maximum number of reconnects allowed
+     *                        If 0 is passed as a value pf the parameter then the default
+     *                        value corresponding configuration parameter will be
+     *                        assumed: Configuration::databaseMaxReconnects().
+     * @param timeoutSec    - (optional) the maximum duration of time allowed for
+     *                        the procedure to wait before a connection will be established.
+     *                        If 0 is passed as a value pf the parameter then the default
+     *                        value corresponding configuration parameter will be
+     *                        assumed: Configuration::databaseConnectTimeoutSec().
+     *
+     * @throws std::invalid_argument - if 'nullptr' is passed in place of 'script'
+     * @throws ConnectError          - failed to establish a connection if connection was
+     *                                 open with fuctory method Connection::open()
+     * @throws ConnectTimeout        - failed to establish a connection within a timeout
+     *                                 if a connection was open with functory method
+     *                                 Connection::openWait()
+     * @throws MaxReconnectsExceeded - multiple failed (due to connecton losses and subsequent reconnects)
+     *                                 attempts to execute the user function. And the number of the attempts
+     *                                 exceeded a limit set by parameter 'maxReconnects'.
+     * @throws Error                 - for any MySQL specific errors. You may also
+     *                                 catch for specific subclasses (other than ConnectError
+     *                                 or ConnectTimeout) of that class if needed.
+     *
+     * @return pointer to the same connector against which the metod was invoked
+     * in case of successful commpletion of the requested operaton.
+     *
+     * @see Configuration::databaseMaxReconnects()
+     * @see Configuration::setDatabaseMaxReconnects()
+     * 
+     * @see Configuration::databaseConnectTimeoutSec()
+     * @see Configuration::setDatabaseConnectTimeoutSec()
+     * 
+     * @see Connection::open()
+     */
+    Connection::Ptr execute(std::function<void(Ptr)> script,
+                            unsigned int maxReconnects=0,
+                            unsigned int timeoutSec=0);
```

Comment:
The first comment is way too generic for this specific context :) Lambdas was a great addition to the language. This mechanism perfectly complements OO by allowing to solve many problems in better ways. It increases visibility of the code which runs once (or from one client). In other cases functions or OO is a good solution. I'm sure people will use `lambdas` more and more over time as they will appreciate them. For me (as someone with wast experience of developing complex projects in JS) this is just given. Others will learnt over time.

Regarding two other comments,  I'm putting a few comments into the header explaining the transaction management and a client's responsibility to thing about side effect. I should also mention that "side/desired effect" is always a responsibility of a programmer regardless if he/she choses to use lambada, regular methods or loop iterations. There is no "silver bullet" (once and for all) solution to this.

Also, improper transactional context (a lack if a transaction or an extra attempt to start/commit/rollback the one will be cause y the database `Connection` implementation). Surely, this will happen at run time. But that is no different from other similar cases. What I have already though I could do here, is to do the transaction management for a client, so that the client would not have to initiate a transaction when calling this method. However, I still need to think about this pattern.